binderhub/tests/test_build.py

```diff
@@ -17,17 +17,17 @@
 
 @pytest.mark.asyncio(timeout=900)
 @pytest.mark.parametrize("slug", [
-    "gh/binderhub-ci-repos/requirements/d687a7f9e6946ab01ef2baa7bd6d5b73c6e904fd",
-    "git/{}/d687a7f9e6946ab01ef2baa7bd6d5b73c6e904fd".format(
-        quote("https://github.com/binderhub-ci-repos/requirements", safe='')
+    "gh/binder-examples/minimal-dockerfile/truly-minimal",
```

Comment:
We should continue to use repos from the binderhub-ci-repos organisation. The example repositories get changed by humans a lot without knowing that changing them might break the CI of binderhub.

Reply:
In particular my request is:
1. to create a `binderhub-ci-repos/minimal-dockerfile` repository
1. use an explicit SHA1 for the tests instead of referring to a branch

Using a fixed hash means that we always get a known input when the tests are executed. A frequent reasons for tests that used to work suddenly not working any more (the famous "tests fail but not because of my changes so please merge with red CI" comment PR authors make) is because an input to our CI has changed.

Reply:
> to create a binderhub-ci-repos/minimal-dockerfile repository

Done in both GitHub and GitLab, using the exact same git content in both repos which makes both have the same SHA etc.

> use an explicit SHA1 for the tests instead of referring to a branch

Done.

> have a test for gh/binderhub-ci-repos/minimal-dockerfile/SHA1, git/quote("binderhub-ci-repos/minimal-dockerfile")/SHA1 and gh/gh/binderhub-ci-repos/minimal-dockerfile/HEAD. Having three builds of the same thing does not use more time than building it once or twice because we have docker layer caching

Done, but note that after the build has been optimized as much as I think is possible, there is a ~65 second penalty per git repo provider. Notes about this are in https://github.com/jupyterhub/binderhub/pull/1202#issuecomment-729046409.